parser/query.go

```diff
@@ -2,14 +2,14 @@ package parser
 
 import (
 	"github.com/vektah/gqlparser/v2/lexer"
-
 	//nolint:revive
 	. "github.com/vektah/gqlparser/v2/ast"
 )
 
 func ParseQuery(source *Source) (*QueryDocument, error) {
 	p := parser{
-		lexer: lexer.New(source),
+		lexer:         lexer.New(source),
+		maxTokenLimit: 15000, // 15000 is the default value
```

Comment:
```suggestion
		maxTokenLimit: 0, // 0 is the default value
```

Reply:
For a large class of existing uses, this is the desired default and does not break backward compatibility, as it allows consumers to opt-in to a different behavior.